### PR TITLE
Adding Reporting to the system

### LINKED ISSUE #9: Generate Locker occupancy by department report
See example of current report: https://docs.google.com/document/d/1GUEzMUo-dKjok0tjhkQcs_xCFW-RivonobhM3PLED6g/edit#heading=h.gr9sc9gf6lrj

### LINKED ISSUE #10: Generate Locker and desk occupancy by location
See original report here: https://docs.google.com/document/d/1GUEzMUo-dKjok0tjhkQcs_xCFW-RivonobhM3PLED6g/edit#heading=h.gr9sc9gf6lrj

### PR DESCRIPTION
**Pull request by carolyncole**
fixes #9
fixes #10

### CODE REVIEW

File: app/views/locker_assignments/assignment_report.csv.erb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Department, Junoirs, Seniors, Grad, Faculty, Staff
```
Comment (christinach) [inline app/views/locker_assignments/assignment_report.csv.erb:1]: Juniors

File: app/models/study_room.rb [hunk 2]
```diff
@@ -34,6 +34,21 @@ def to_s
...
+  end
+
+  def space_assigned_totals
+    @space_assigned_totals ||= StudyRoomAssignment.where(released_date: nil).joins(:study_room)
+                                                  .group(:general_area).order(:general_area).count
+  end
+
+  def space_report
+    space_totals.map do |key, total|
+      [key, [total, total, space_assigned_totals[key], total - space_assigned_totals[key]]]
```
Comment (christinach) [inline app/models/study_room.rb:48]: why there are two 'total' ?

Comment (christinach) [reply app/models/study_room.rb:48]: should the second total be a different calculation?

Comment (carolyncole) [reply app/models/study_room.rb:48]: No, with Study rooms they do not disable them.  That is just a locker construct.

**Review by christinach — APPROVED** [review]

**Pull request merged**